README.md

```diff
@@ -345,3 +345,72 @@ w := kafka.NewWriter(kafka.WriterConfig{
 	Dialer:   dialer,
 })
 ```
+
+## SASL Support
+
+For a bare bones Conn type or in the Reader/Writer configs you can specify a dialer option for SASL authentication. If the `SASLMechanism` field is `nil`, it will not authenticate with SASL.
```

Comment:
What about something like this?
```suggestion
You can specify an option on the `Dialer` to use SASL authentication.  The `Dialer` can be used directly to open a `Conn` or it can be passed to a `Reader` or `Writer` via their respective configs.  If the `SASLMechanism` field is `nil`, it will not authenticate with SASL.
```
I think it aligns a little bit more with the code since, in all cases, you configure the mechanism on the dialer.

Definitely open to further edits. 😄